File: app/src/main/java/com/mmutert/freshfreezer/ui/databinding/IntStringConverter.java
package com.mmutert.freshfreezer.ui.databinding;

import androidx.databinding.InverseMethod;


public class IntStringConverter {

    @InverseMethod("stringToInt")
    public static String intToString(int toConvert) {
        return String.valueOf(toConvert);
    }

    public static int stringToInt(String toConvert) {
        try {
            return Integer.parseInt(toConvert);
        } catch (NumberFormatException e) {
            return 0;
        }
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/ui/databinding/IntStringConverter.kt
package com.mmutert.freshfreezer.ui.databinding

import androidx.databinding.InverseMethod

object IntStringConverter {
    @InverseMethod("stringToInt")
    fun intToString(toConvert: Int): String {
        return toConvert.toString()
    }

    fun stringToInt(toConvert: String): Int {
        return try {
            toConvert.toInt()
        } catch (e: NumberFormatException) {
            0
        }
    }
}